modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -784,6 +784,11 @@
   [_]
   (CurrentMomentForm. nil))
 
+(defmethod sql.qp/->honeysql [:bigquery-cloud-sdk :now]
+  [driver _clause]
+  (->> (sql.qp/current-datetime-honeysql-form driver)
+       (->temporal-type :timestamp)))
```

Comment:
I didn't implement this in `sql.qp/current-datetime-honeysql-form` because bigquery's implementation has a special ability to change temporal type without casting, that I think is worth keeping around for future optimizations. `:now` can't take advantage from this yet because it could be an argument to :convert-timezone, which expects a typed honeysql form. See the tests for https://github.com/metabase/metabase/blob/a056da0f42ac689af247be747323f0445e01a9d8/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk/query_processor_test.clj#L597